packages/shared/lib/services/connection.service.ts

```diff
@@ -522,26 +524,42 @@ class ConnectionService {
         await trx.from<StoredConnection>(`_nango_connections`).whereIn('id', ids).andWhere({ deleted: false }).update({ metadata });
     }
 
-    public async replaceConnectionConfig(connection: Connection, config: ConnectionConfig) {
-        await db.knex
+    public async replaceConnectionConfig(connection: Connection, config: ConnectionConfig, trx?: Knex.Transaction) {
+        const query = db.knex
```

Comment:
usually you can swap the object; it's annoying that knex is not exactly the same type as trx but it works almost transparently 
```ts
    public async replaceConnectionConfig(connection: Connection, config: ConnectionConfig, trx?: Knex.Transaction) {
        const query = (trx || db.knex)
```